### PR TITLE
feat: Add profit multiplier

### LINKED ISSUE #57: Add a profit multiplier/percentage to batch relaying
## Summary

Validators don't have an easy way to set how much they want to earn. They should be able to add a % of profit on top of `fees - profits`. This will provide more control over the orchestrator, add some competition, and improve incentives.

## Proposal

Add a flag like `--relayer-profit-mult` or `--relayer-profit-percentage` that receives a number. Then profitability will be calculated using that value so the batch will be profitable only if batch fees are over X% greater than gas cost. I'm not very good at math, I'll need @alexanderbez to help me define this. 

----

## For Admin Use

- [ ] Not duplicate issue
- [ ] Appropriate labels applied
- [ ] Appropriate contributors tagged
- [ ] Contributor assigned/self-assigned

### PR DESCRIPTION
**Pull request by facundomedica**
Added the flag `--relayer-profit-multiplier` which defaults to 1.1 meaning that the amount of fees paid by a batch must be at least 1.1 times the estimated gas cost of the ETH tx to be relayed successfully.

- With a profit multiplier of 1.0, the orchestrator will relay batches at cost.
- With a profit multiplier of 0.0, the orchestrator will relay every batch without checking fees or gas cost.

`totalFee > (gasCost * profitMultiplier)`

Closes #57

### CODE REVIEW

File: cmd/peggo/flags.go [hunk 1]
```diff
@@ -10,39 +10,40 @@ const (
...
+	flagRelayBatches            = "relay-batches"
+	flagCoinGeckoAPI            = "coingecko-api"
+	flagEthGasPrice             = "eth-gas-price"
+	flagEthGasLimit             = "eth-gas-limit"
+	flagPowerThreshold          = "power-threshold"
+	flagAutoApprove             = "auto-approve"
+	flagOrchLoopDuration        = "orch-loop-duration"
+	flagEthBlocksPerLoop        = "eth-blocks-per-loop"
+	flagEthPendingTXWait        = "eth-pending-tx-wait"
+	flagRelayerProfitMultiplier = "relayer-profit-multiplier"
```
Comment (alexanderbez) [inline cmd/peggo/flags.go:46]: nit: wdyt? 

```suggestion
	flagProfitMultiplier = "profit-multiplier"
```

File: cmd/peggo/orchestrator.go [hunk 2]
```diff
@@ -224,6 +225,7 @@ func getOrchestratorCmd() *cobra.Command {
 	cmd.Flags().String(flagCoinGeckoAPI, "https://api.coingecko.com/api/v3", "Specify the coingecko API endpoint")
 	cmd.Flags().Duration(flagEthPendingTXWait, 20*time.Minute, "Time for a pending tx to be considered stale")
 	cmd.Flags().String(flagEthAlchemyWS, "", "Specify the Alchemy websocket endpoint")
+	cmd.Flags().Float64(flagRelayerProfitMultiplier, 1.1, "Multiplier to apply to relayer profit")
```
Comment (alexanderbez) [inline cmd/peggo/orchestrator.go:228]:
```suggestion
	cmd.Flags().Float64(flagRelayerProfitMultiplier, 1.0, "Multiplier to apply to relayer profit")
```

File: orchestrator/ethereum/peggy/peggy_contract.go [hunk 3]
```diff
@@ -253,6 +253,10 @@ func (s *peggyContract) GetERC20Decimals(
 		return 0, errors.Wrap(err, "ERC20 'decimals' call failed")
 	}
 
+	if s.erc20DecimalCache == nil {
```
Comment (alexanderbez) [inline orchestrator/ethereum/peggy/peggy_contract.go:256]: good catch, I forgot. Thanks

File: orchestrator/relayer/batch_relaying.go [hunk 4]
```diff
@@ -242,16 +245,22 @@ func (s *peggyRelayer) IsBatchProfitable(
 	}
 
 	usdTokenPriceDec := decimal.NewFromFloat(usdTokenPrice)
-	// decimals (uint8) can be safely casted into int32 because the max uint8 is 255 and the max int32 is 2147483647
+	// Decimals (uint8) can be safely casted into int32 because the max uint8 is 255 and the max int32 is 2147483647.
 	totalFeeInUSDDec := decimal.NewFromBigInt(totalBatchFees, -int32(decimals)).Mul(usdTokenPriceDec)
 
+	// Simplified: totalFee > (gasCost * profitMultiplier).
+	isProfitable := totalFeeInUSDDec.GreaterThan(gasCostInUSDDec.Mul(decimal.NewFromFloat(profitMultiplier)))
```
Comment (alexanderbez) [inline orchestrator/relayer/batch_relaying.go:252]: shouldn't this be >=?

**Review by alexanderbez — COMMENTED** [review]
Looks great! Just left a few comments 👍

File: cmd/peggo/flags.go (hunk 1 above)
Comment (facundomedica) [reply cmd/peggo/flags.go:46]: Agree, there's no other profit to get confused about. I'll change that 👌

File: orchestrator/relayer/batch_relaying.go (hunk 4 above)
Comment (facundomedica) [reply orchestrator/relayer/batch_relaying.go:252]: Right, changed it to `GreaterThanOrEqual`

**Review by alexanderbez — APPROVED** [review]
🎉

**Review by alexanderbez — APPROVED** [review]

**Pull request merged**